keras/src/callbacks/orbax_checkpoint.py

```diff
@@ -0,0 +1,579 @@
+        return False
+
+    def _get_current_step(self):
+        # A reliable way to get a global step count
+        # Using optimizer iterations is common
+        if hasattr(self.model, "optimizer") and hasattr(
+            self.model.optimizer, "iterations"
+        ):
+            # Convert potential backend tensor to int
+            return int(
+                backend.convert_to_numpy(self.model.optimizer.iterations)
+            )
+        else:
+            # Fallback: use global batch count
+            return self._total_batches_seen
+
+    def _save_checkpoint(self, step, logs=None):
+        """Save a checkpoint at the given step."""
+        if self.model is None:
+            return
+
+        # --- Prepare Composite State (Backend-Agnostic) ---
+        model_weights_np, optimizer_vars_np = _get_state_as_numpy(self.model)
+
+        if model_weights_np is None:
+            if self.verbose > 0:
+                print("OrbaxCheckpoint: Skipping save due to conversion error")
+            return
+
+        composite_state = {"model_weights": model_weights_np}
+        if self.save_optimizer_state and optimizer_vars_np is not None:
+            composite_state["optimizer_state"] = optimizer_vars_np
+
+        # Add metrics state if specified
+        if self.save_metrics_state and hasattr(self.model, "metrics"):
+            metrics_vars_np = []
+            for metric in self.model.metrics:
+                if hasattr(metric, "variables") and metric.variables:
+                    # Convert metric variables to numpy
+                    metric_vars = [
+                        backend.convert_to_numpy(var)
+                        for var in metric.variables
+                    ]
+                    metrics_vars_np.append(metric_vars)
+
+            if metrics_vars_np:
+                composite_state["metrics_state"] = metrics_vars_np
+
+        # Add metadata if specified
+        if self.save_metadata is not None:
+            if callable(self.save_metadata):
+                metadata = self.save_metadata(self._current_epoch, logs)
+            else:
+                metadata = self.save_metadata
+            if metadata:
+                composite_state["metadata"] = metadata
+
+        # Add data iterator state if specified
+        if self.save_data_iterator is not None:
+            if callable(self.save_data_iterator):
+                iterator_state = self.save_data_iterator(
+                    self._current_epoch, logs
+                )
+            else:
+                iterator_state = self.save_data_iterator
+            if iterator_state:
+                composite_state["data_iterator"] = iterator_state
+
+        # --- Save Logic ---
+        # Assuming single host or JAX backend with jax.distributed initialized
+        # for now.
+        # A robust implementation would need a backend-aware way to check
+        # process_index.
+        is_primary_host = backend.get_process_index() == 0
+
```

Comment:
Wait, I thought `backend.get_process_index()` is the robust way to do it.

That should work in multi-host setups, right? What is missing?